<a href="https://colab.research.google.com/github/davicn/ada_logica_programacao/blob/main/aula_3.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Aula 3

## Laços de repetição condicionais (WHILE)

![image info](https://cdn.programiz.com/sites/tutorial2program/files/whileLoopFlowchart.jpg)


In [ ]:
while True:
    print('teste')

A saída de streaming foi truncada nas últimas 5000 linhas.
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste
teste

KeyboardInterrupt: ignored

In [ ]:
cont = 0

while cont<=250:
    print(cont)
    cont = cont + 1

0
1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88
89
90
91
92
93
94
95
96
97
98
99
100
101
102
103
104
105
106
107
108
109
110
111
112
113
114
115
116
117
118
119
120
121
122
123
124
125
126
127
128
129
130
131
132
133
134
135
136
137
138
139
140
141
142
143
144
145
146
147
148
149
150
151
152
153
154
155
156
157
158
159
160
161
162
163
164
165
166
167
168
169
170
171
172
173
174
175
176
177
178
179
180
181
182
183
184
185
186
187
188
189
190
191
192
193
194
195
196
197
198
199
200
201
202
203
204
205
206
207
208
209
210
211
212
213
214
215
216
217
218
219
220
221
222
223
224
225
226
227
228
229
230
231
232
233
234
235
236
237
238
239
240
241
242
243
244
245
246
247
248
249
250


## Faça um programa que imprima na tela apenas os números ímpares entre 1 e 50.


In [ ]:
numImpar = 1

while numImpar <= 48:
    print(numImpar)
    numImpar +=2
print(numImpar)

1
3
5
7
9
11
13
15
17
19
21
23
25
27
29
31
33
35
37
39
41
43
45
47
49


In [ ]:
cont = 1
while cont <= 50:
  print(cont)
  cont = cont + 2

1
3
5
7
9
11
13
15
17
19
21
23
25
27
29
31
33
35
37
39
41
43
45
47
49


In [ ]:
numero = 0

while numero <= 50:
    if numero % 2 == 0:
         continue
    print(numero)

KeyboardInterrupt: ignored

In [ ]:
numero=0

while numero<=50:
  if numero%2 == 1:
      print(numero)
  numero += 1 # numero = numero + 1


send_msg()

1
3
5
7
9
11
13
15
17
19
21
23
25
27
29
31
33
35
37
39
41
43
45
47
49


## Faça um programa que calcule o fatorial de um número inteiro fornecido pelo usuário

Ex.: 5!=5.4.3.2.1 = 120.

In [ ]:
entrada= (int(input('Digite um número')))

contador = 1
fatorial = entrada

while (contador < entrada):
   fatorial = (fatorial * contador) # fatorial *= contador
   contador = (contador + 1) # contador += 1

print(fatorial)

Digite um número10
3628800


In [ ]:
num = 5 #int(input('Digite um número'))

fatorial = 1

while num > fatorial:
  num -= 1
  resul *=  num #num * (num -=1)
  print(resul)

NameError: ignored

In [ ]:
3 * 2 * 1

6

In [ ]:
10! = 10 * 9 * 8 * 7 ... * 1

##  Comandos de manipulação de fluxo (BREAK/CONTINUE)

In [ ]:
cont = 1

while True:
    print(cont) 
    cont += 1

    if cont == 11:
        print('acabou!!!')
        print('acabou amesmo!!!')
        break

1
2
3
4
5
6
7
8
9
10
acabou!!!
acabou amesmo!!!


In [ ]:
numero_secreto = 42

contador = 0

while contador < 10:
  tentativa = int(input('Adivinhe o número secreto: '))
  contador += 1
  if tentativa != numero_secreto:
    print('Errou!')
    continue
  print('Acertou!')
  break
else:
  print('Acabaram as tentativas. Você perdeu.')

Adivinhe o número secreto: 42
Acertou!


## Faça um programa que peça uma nota, entre zero e nove Mostre uma mensagem caso o valor seja inválido e continue pedindo até que o usuário informe um valor válido.

In [ ]:

cont = 
while 
nota = int(input('Digite uma nota: '))


In [ ]:
cpf = '004130950' 

'004130950'

In [ ]:
nota = int(input('Digite uma nota entre zero e nove:  '))

while 0 >= nota <= 9:
  print('Nota válida!')
  break
else:
  print('Número errado, tente novamente')
  nota = int(input('Digite uma nota entre zero e nove:  '))
  

Digite uma nota entre zero e nove:  100
Número errado, tente novamente
Digite uma nota entre zero e nove:  12


In [ ]:
while True:
  nota = float(input("Informe a nota: "))
  if 1 <= nota <= 9:
    print ("Nota válida.")
    break
  else:
    print ("Nota inválida.")
    continue

Informe a nota: 12
Nota inválida.
Informe a nota: 100
Nota inválida.
Informe a nota: 13
Nota inválida.
Informe a nota: 2
Nota válida.


In [ ]:
num = int(input('digite uma nota: '))

while num >  9:
  num = int(input('digite uma nota: '))
  continue
if num <= 9:
  print('válido')

digite uma nota: 9
válido


In [ ]:
GIT -> ferramento, em linha de comandom para versionamento de códigos

GITHUB - > provedor git
Gitlab
Bitbucket